source/extensions/common/wasm/wasm.cc

```diff
@@ -2045,6 +2045,23 @@ std::string Plugin::makeLogPrefix() const {
   return prefix;
 }
 
+Context::~Context() {
+  // Cancel any outstanding requests.
+  for (auto& p : http_request_) {
+    p.second.request->cancel();
+  }
+  for (auto& p : grpc_call_request_) {
+    p.second.request->cancel();
+  }
+  for (auto& p : grpc_stream_) {
+    p.second.stream->resetStream();
```

Comment:
Are these all synchronous? If it posts back to libevent, the this won't help.

Reply:
@PiotrSikora Isn't google gRPC still running in a separate silo and cancellation is async? I don't remember what happened to it afterwards, but it used to run on a separate worker thread.

Reply:
I checked gRPC code. LG.